### PR TITLE
Improve Connections

### PR DESCRIPTION
**Pull request by juliangieseke**
this PR improves the connections "here and there" :)

### CODE REVIEW

File: src/AbstractConnection.js [hunk 1]
```diff
@@ -121,10 +133,14 @@ export default class AbstractConnection extends EventEmitter {
    * - DATA everytime you get data from the server
    * - ERROR, ABORT, _or_ COMPLETE after the Connection is closed
    */
-  open() {}
+  open() {
+    this.opened = Date.now();
```
Comment (DanielMSchmidt) [inline src/AbstractConnection.js:137]: This requires the extending connection to call `super()` right?

File: src/XHRConnection.js [hunk 2]
```diff
@@ -35,18 +35,24 @@ export default class XHRConnection extends AbstractConnection {
         Accept: "application/json"
       },
       responseType = "json",
-      timeout = 0
+      timeout = 0,
+      listeners = [],
```
Comment (DanielMSchmidt) [inline src/XHRConnection.js:39]: This seems odd to me, having addListeners introduced in the "parent" and making the setting and handling of listeners in the child. Can we "just" put them into the super call and move all of this logic on the bottom of the constructor in the abstract connection?

File: src/XHRConnection.js [hunk 3]
```diff
@@ -35,18 +35,24 @@ export default class XHRConnection extends AbstractConnection {
...
+      open = false
```
Comment (DanielMSchmidt) [inline src/XHRConnection.js:40]: This and listeners is missing in the documentation comment

File: src/XHRConnection.js [hunk 4]
```diff
@@ -103,6 +127,11 @@ export default class XHRConnection extends AbstractConnection {
      * @name XHRConnection#xhr
      */
     Object.defineProperty(this, "xhr", { value: new XMLHttpRequest() });
+
+    // open immediatly?
```
Comment (DanielMSchmidt) [inline src/XHRConnection.js:131]: I think this comment doesn't help so much, but confuses me a bit. Why the "?" ?

File: src/XHRConnection.js [hunk 5]
```diff
@@ -103,6 +127,11 @@ export default class XHRConnection extends AbstractConnection {
...
+    if(options.open) {
+      this.open();
```
Comment (DanielMSchmidt) [inline src/XHRConnection.js:133]: Aren't the headers missing here?

File: src/XHRConnection.js [hunk 6]
```diff
@@ -127,80 +156,111 @@ export default class XHRConnection extends AbstractConnection {
...
 
     this.xhr.open(this.method, this.url);
     this.xhr.timeout = this.timeout;
     this.state = XHRConnection.OPEN;
-    this.emit(
-      ConnectionEvent.OPEN,
-      new ConnectionEvent(this, ConnectionEvent.OPEN)
-    );
+    this.opened = Date.now();
+    window.setTimeout(() => {
```
Comment (DanielMSchmidt) [inline src/XHRConnection.js:239]: Can we move the setTimeout to the callee side (so `emit`)? It seems pretty repetitive, like a good default and easy to forget

File: package.json [hunk 7]
```diff
@@ -18,9 +18,9 @@
   },
   "devDependencies": {
     "babel-cli": "^6.26.0",
-    "babel-jest": "^21.2.0",
+    "babel-jest": "22.4.3",
```
Comment (DanielMSchmidt) [inline package.json:21]: I think we should either nail every dependency or use carrots all the way, this seems inconcise

**Review by DanielMSchmidt — CHANGES_REQUESTED** [review]
Added some comments :)

File: src/AbstractConnection.js (hunk 1 above)
Comment (juliangieseke) [reply src/AbstractConnection.js:137]: either calling super or copying the line, yes. 

not sure if we need these metrics at all tho…

File: src/XHRConnection.js (hunk 2 above)
Comment (juliangieseke) [reply src/XHRConnection.js:39]: `addListerners` isnt introduced in `AbstractConnection`?!

**Comment by juliangieseke** [comment]
uh, i need to get this updated at some point 🤔

**Comment by DanielMSchmidt** [comment]
> uh, i need to get this updated at some point 

No update?

**Comment by nLight** [comment]
I'm calling in a meeting to align 🥹

**Comment by juliangieseke** [comment]
Was just cleaning up my PR list 🙈

I think its save to say these improvements wont be necessary anymore.

**Pull request closed without merging**